Bot/Services/DatabaseHandler.cs

```diff
@@ -0,0 +1,113 @@
+namespace NetBot.Bot.Services
+{
+    public class DatabaseHandler
+    {
+        private static readonly ILog log = LogManager.GetLogger(typeof(DatabaseHandler));
+        private static readonly string connectionString = DotNetEnv.Env.GetString("CONNECTION_STRING") + "Min Pool Size=5;";
+
+        public static async Task<bool> CheckDNDTos(SocketSlashCommand command)
+        {
+            ulong userID = command.User.Id;
+
+            string sql = "SELECT agreeToDND FROM users WHERE id = :userID";
+            using (OracleConnection connection = new OracleConnection(connectionString))
+            using (OracleCommand getTos =  new OracleCommand(sql, connection))
+            {
+                try
+                {
+                    await CheckCreateUser(userID);
+
+                    await connection.OpenAsync();
+
+                    getTos.Parameters.Add("userID", (long)userID);
+                    short? tos = (short?)await getTos.ExecuteScalarAsync(); // When reading from the database, it's a short?; when writing it's an int. Very confusing.
+
+                    if (tos == 0)
+                    {
+                        Embed embed = new EmbedBuilder()
+                            .WithTitle("Hold up!")
+                            .WithDescription("Before you can use any commands related to DND, you must aknowledge that you already own the rulebooks, and otherwise any content that goes with it. If you agree to this, run the command </agreetoterms:1106074501122367538>.")
+                            .WithColor(Color.Red)
+                            .Build();
+                        await command.RespondAsync(embed: embed, ephemeral: true);
+                        return false;
+                    }
+                    else return true;
+                } 
+                catch (Exception ex)
+                {
+                    log.Error(ex.Message, ex);
+                    return true;
+                }
+            }
+        }
+
+        public static async Task CheckCreateUser(ulong id)
+        {
+            string sql = "SELECT * FROM users WHERE id = :userID";
+
+            using (OracleConnection connection = new(connectionString))
+            using (OracleCommand command = new(sql, connection))
+            {
+                await connection.OpenAsync();
+                command.Parameters.Add(new OracleParameter("userID", (long)id));
+                var result = await command.ExecuteScalarAsync();
+                if (result is null)
+                {
+                    OracleCommand createUser = new("INSERT INTO users (id) VALUES (:userID)", connection);
+                    createUser.Parameters.Add("userID", (long)id);
+                    await createUser.ExecuteNonQueryAsync();
+                }
+            }
+        }
+
+        public static async Task<(bool? previous, bool? result)> UpdateDNDTerms(ulong discordId, bool value)
+        {
+            int agreeToTerms = value ? 1 : 0;
```

Comment:
What's the point in using an int as opposed to a bool for the db ? The performance save is negligible and this way I feel can just cause further confusion down the line when some things are `true` or `false` and others are `1` or `0`